docs/ha-considerations.md

```diff
@@ -0,0 +1,337 @@
+# High Availability Considerations
+
+## Overview
+
+When setting up a productive cluster, high availability, i.e. the cluster's ability to remain operational even if some control plane or worker nodes fail, is usually a requirement. For worker nodes, assuming that there are enough of them, this is part of the very cluster functionality; however redundancy of control plane nodes and `etcd` instances needs to be catered for when planning and setting up a cluster.
```

Comment:
productive -> production
as users are more used to the later.

---

> high availability, i.e. the cluster's ability to remain operational even if some control plane or worker nodes fail,

i'd change this to:

> high availability (the cluster's ability to remain operational even if some control plane or worker nodes fail), ...

k/website recommends to not use latin abbreviations in docs, although we don't follow this strictly on our side.

---

`functionality; however redundancy` -> `functionality. However redundancy`